src/xref2/subst.ml

```diff
@@ -199,7 +202,9 @@ let rec resolved_module_path :
   | `Hidden p1 -> `Hidden (resolved_module_path s p1)
   | `Canonical (p1, p2) ->
       `Canonical (resolved_module_path s p1, module_path s p2)
-  | `OpaqueModule m -> `OpaqueModule (resolved_module_path s m)
+  | `OpaqueModule m ->
+      if s.unresolve_opaque_paths then raise Invalidated
```

Comment:
This exception is raised and caught in a lot of places, what about turning it into a value to ensures the new raises are handled ? (like `or_replaced`)

Reply:
Yes, we should do this. The mechanism pre-exists though, so I think we can do it in a follow-up PR.